src/combinator.rs

```diff
@@ -73,12 +73,10 @@ where
 }
 
 /// Apply combinator, automatically converts between errors if the underlying type supports it
-pub fn upgrade_error<I, O, E1: ParseError<I>, E2: ParseError<I>, F>(
-    mut f: F,
-) -> impl FnMut(I) -> IResult<I, O, E2>
+pub fn upgrade_error<I, O, E1: ParseError<I>, E2, F>(mut f: F) -> impl FnMut(I) -> IResult<I, O, E2>
 where
     F: FnMut(I) -> IResult<I, O, E1>,
-    E2: From<E1>,
+    E2: ParseError<I> + From<E1>,
```

Comment:
I understand the clippy warning, but I'm surprised it doesn't complain more generally that some bounds are defined in the function generics, and some are in the where clause.

Reply:
I ended up doing this in two separate added commits: one to make the location of the bounds consistent, and then one to sort the bounds in the order of the generic parameters.

Reply:
I see there are later commits, marking this as resolved. Thanks!